lib/Discovery_Schema.cpp

```
@@ -2241,3 +2335,676 @@ int Discovery_Schema::log_query_tool_call(
+	result->add_column_definition(SQLITE_TEXT, "OK_msg");
+	result->add_column_definition(SQLITE_TEXT, "log");
+	result->add_column_definition(SQLITE_TEXT, "apply");
+	result->add_column_definition(SQLITE_TEXT, "comment");
+
+	pthread_rwlock_rdlock(&mcp_rules_lock);
+
+	for (size_t i = 0; i < mcp_query_rules.size(); i++) {
+		MCP_Query_Rule* rule = mcp_query_rules[i];
+		char** pta = (char**)malloc(sizeof(char*) * 17);
+
+		pta[0] = strdup(std::to_string(rule->rule_id).c_str());           // rule_id
```

Cppcheck (2.19.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)